app/pages/home/home.tsx

```diff
@@ -4,36 +4,45 @@ import { Flex, Box } from '@blockstack/ui';
 import { ChainID } from '@blockstack/stacks-transactions';
 import { deriveRootKeychainFromMnemonic, deriveStxAddressChain } from '@blockstack/keychain';
 
-import { selectMnemonic } from '../../store/keys';
-import { BIP32Interface } from '@blockstack/keychain/node_modules/bip32';
+import { selectMnemonic, selectKeysSlice } from '../../store/keys';
+import { BIP32Interface } from '../../types';
 
 //
 // Placeholder component
 export const Home: React.FC = () => {
   const mnemonic = useSelector(selectMnemonic);
-  const [keychain, setKeychain] = useState<{ rootNode: BIP32Interface } | null>(null);
-  useEffect(() => {
-    const deriveMasterKeychain = async () => {
-      if (!mnemonic) return;
-      const resp = await deriveRootKeychainFromMnemonic(mnemonic, '');
-      setKeychain(resp);
-    };
-    void deriveMasterKeychain();
-  }, [mnemonic]);
+  const keys = useSelector(selectKeysSlice);
+  // const [keychain, setKeychain] = useState<{ rootNode: BIP32Interface } | null>(null);
+  // useEffect(() => {
+  //   const deriveMasterKeychain = async () => {
+  //     if (!mnemonic) return;
+  //     const resp = await deriveRootKeychainFromMnemonic(mnemonic, '');
+  //     setKeychain(resp);
+  //   };
+  //   void deriveMasterKeychain();
+  // }, [mnemonic]);
 
-  const { privateKey } = keychain?.rootNode
-    ? deriveStxAddressChain(ChainID.Testnet)(keychain.rootNode)
-    : { privateKey: '' };
+  // if (keychain === null) return <div>Homepage, but no keychain can be derived</div>;
 
-  if (!mnemonic) return <>'How you get to homepage without a mnemonic?'</>;
+  // const rootNode = deriveStxAddressChain(ChainID.Testnet)(keychain.rootNode);
 
-  console.log(keychain);
+  // const privateKey = rootNode.privateKey;
+
+  // const base58 = rootNode.childKey.toBase58();
+
+  if (!mnemonic) return <>How you get to homepage without a mnemonic?</>;
+
+  // console.log(keychain);
 
   return (
     <Flex pt="120px" flexDirection="column" mx="loose">
       <Box>Mnemonic: {mnemonic}</Box>
-      <Box mt="loose">MnemonicEncryptedHex: {(keychain as any)?.encryptedMnemonicHex}</Box>
-      <Box mt="loose">Private key: {privateKey}</Box>
+      {/* <Box mt="loose">MnemonicEncryptedHex: {privateKey}</Box> */}
+      {/* <Box mt="loose">Private key: {privateKey}</Box> */}
+      {/* <Box mt="loose">Base58: {base58}</Box> */}
+      <Box mt="loose">Salt: {(keys as any).salt}</Box>
+      <Box mt="loose">Password: {(keys as any).password}</Box>
+      <Box mt="loose">Stretched Key: {(keys as any).derivedEncryptionKey}</Box>
     </Flex>
```

Comment:
This can be achieved a bit cleaner with `Stack`

```tsx
<Stack spacing="loose">
... children
</Stack>
```

Reply:
Noted! This component will completely change though, these values are just to test it's working.